File: Frontend/src/modelos/paciente.ts
export interface Paciente {
  id: number;
  nombre: string;
  apellido: string;
  // Nullable: un paciente autogestionado (alta por DNI) no completa estos
  // datos hasta pedir su primer turno; el que carga el Administrador
  // manualmente sí los tiene (salvo email, que es exclusivo del autoservicio).
  telefono: string | null;
  obraSocial: string | null;
  email: string | null;
  // Solo tiene valor para pacientes autogestionados.
  dni: string | null;
}

// El alta manual del Administrador sigue pidiendo estos 4 campos. El DNI es
// opcional y solo se muestra/envía al editar (permite corregir uno mal
// cargado por un paciente autogestionado); vacío lo limpia.
export interface GuardarPacienteDto {
  nombre: string;
  apellido: string;
  telefono: string;
  obraSocial: string;
  dni: string;
}

// DTO acotado para el autoservicio del paciente (no permite tocar
// nombre/apellido/DNI, a diferencia de GuardarPacienteDto).
export interface ActualizarContactoPacienteDto {
  telefono: string;
  obraSocial: string;
  email: string;
}

File: Backend/Clinica.Dominio/Entidades/Paciente.cs
namespace Clinica.Dominio.Entidades;

public class Paciente
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Apellido { get; set; } = string.Empty;

    // Solo se completa cuando el paciente se autogestionó (ver
    // ServicioAutenticacion.AccederComoPacienteAsync); un paciente cargado
    // manualmente por el Administrador no tiene login propio y queda null.
    public string? Dni { get; set; }

    // Nullable porque el alta autogestionada del paciente (solo Nombre +
    // Apellido + DNI) no los pide; el alta manual del Administrador sí los
    // sigue exigiendo a través de su propio validador.
    public string? Telefono { get; set; }
    public string? ObraSocial { get; set; }
    public string? Email { get; set; }

    public DateTime FechaCreacion { get; set; }

    public ICollection<Turno> Turnos { get; set; } = new List<Turno>();
}
